src/structures/sound_action.rs

```diff
@@ -118,7 +118,10 @@ impl SoundAction {
                         MusicDownloadStatus::NotDownloaded
                     },
                 );
-                player.list.insert(player.current + 1, video);
+                match player.list.len() {
```

Comment:
You can replace that with an if .is_empty()

Reply:
Yes, thanks I think it would be more wise because of condition simplicity.